app/src/main/java/net/skyscanner/backpack/demo/compose/AppSearchModalStory.kt

```diff
@@ -0,0 +1,204 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.demo.compose
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import net.skyscanner.backpack.compose.appsearchmodal.Action
+import net.skyscanner.backpack.compose.appsearchmodal.AppSearchModalResult
+import net.skyscanner.backpack.compose.appsearchmodal.BpkAppSearchModal
+import net.skyscanner.backpack.compose.appsearchmodal.Item
+import net.skyscanner.backpack.compose.appsearchmodal.Section
+import net.skyscanner.backpack.compose.appsearchmodal.SectionHeading
+import net.skyscanner.backpack.compose.appsearchmodal.Shortcut
+import net.skyscanner.backpack.compose.button.BpkButton
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.tokens.Airports
+import net.skyscanner.backpack.compose.tokens.City
+import net.skyscanner.backpack.compose.tokens.Landmark
+import net.skyscanner.backpack.compose.tokens.UseLocation
+import net.skyscanner.backpack.demo.R
+import net.skyscanner.backpack.demo.components.AppSearchModalComponent
+import net.skyscanner.backpack.demo.meta.ComposeStory
+
+@Composable
+@AppSearchModalComponent
+@ComposeStory("Content")
+fun AppSearchModalStoryContent(modifier: Modifier = Modifier) {
+    AppSearchModalStory(result = contentResult)
+}
+
+@Composable
+@AppSearchModalComponent
+@ComposeStory("Loading")
+fun AppSearchModalStoryLoading(modifier: Modifier = Modifier) {
+    AppSearchModalStory(result = loadingResult)
+}
+
+@Composable
+@AppSearchModalComponent
+@ComposeStory("Error")
+fun AppSearchModalStoryError(modifier: Modifier = Modifier) {
+    AppSearchModalStory(result = errorResult)
+}
+
+@Composable
+private fun AppSearchModalStory(
+    result: AppSearchModalResult,
+    modifier: Modifier = Modifier,
+) {
+    Column(modifier.fillMaxSize()) {
+        DefaultAppSearchModalSample(result = result)
+    }
+}
+
+@Composable
+internal fun DefaultAppSearchModalSample(
+    result: AppSearchModalResult,
+    modifier: Modifier = Modifier,
+) {
+    val showModal = rememberSaveable { mutableStateOf(true) }
+    Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+        BpkButton(
+            text = stringResource(R.string.generic_show),
+            onClick = { showModal.value = true },
+        )
+    }
+    if (showModal.value) {
+        BpkAppSearchModal(
+            modifier = modifier,
+            title = "Destination",
```

Comment:
can you use string resourced please? we can probably simplify by just having one copy for each item for example as the content doesn't matter much

Reply:
done